### PR TITLE
Refocus send textarea after message deletion

### PR DESCRIPTION
**Pull request by deciare**
Recofus the send textarea after deleting messages using the menu, `/delmode`, or `/del` (or cancelling out of deleting messages).

This makes focus behaviour consistent with other kinds of deletion (such as deleting swipes or chats).

### CODE REVIEW

**Comment by Wolfsblvt** [comment]
I think this should be made an optional parameter for that slash command. Or maybe a whole separate command in itself.

Otherwise you'll have side effects with all the script where those two commands are just part of the bigger script, might be executed multiple times, etc etc. It'll focus while the script is still running. Not sure how much that makes sense.

**Comment by deciare** [comment]
I was trying to think of things that this change in focus behaviour could possibly break, and was drawing a blank. I wasn't thinking about scripts, though.

Very simple test case:
```
/del 1 | /input "testing"
```

Without the changes in this PR, the textarea in "testing" pop-up is focused. With the changes in this PR, the textarea in "testing" briefly receives focus but then loses it.

I wonder if there's a way to automatically determine whether whether the send textarea should be refocused based on whether something else has taken focus after entering but before leaving delete mode.

**Comment by Wolfsblvt** [comment]
I think we should consult our resident expert @LenAnderson here, he might have the most valuable input.

I fear it's either as I said, has to be an optional parameter that has to be explicitly set, or a focus command itself. Checking whether the command is the last in the chain isn't really something that looks consistent enough to be a feature that makes sense.

The more I think about it, the more I feel like having a `/focus-input` command is the way to go.

**Comment by deciare** [comment]
I frequently delete messages using `/del 1` and so forth, and still manage to be caught off-guard every time the send field loses focus. Having a separate command that refocuses the input field would defeat the purpose of what I'm hoping to change.

I don't have a better solution yet (that doesn't involve adding an option). Still thinking about it.

**Comment by deciare** [comment]
I changed the behaviour so that the send textarea is only focused if the element that was focused at the time the Cancel or Delete button was clicked has not changed.

Using the same test case as above, the pop-up text field now remains focused (if one exists), and the send textarea is refocused otherwise.

**Comment by Wolfsblvt** [comment]
I think that could work. Nice. Maybe Cohee or Len has a second opinion.

**Comment by LenAnderson** [comment]
Too tired to check this in detail right now, but you're assigning the initial element after pointerdown. Which means it's likely to be always body.
While that might help if something else receives focus before the callback reaches the point where it focuses the input, it doesn't really deal with cases where the chat input wasn't the focused element in the first place. Chat input isn't the only place to run scripts from after all. And if the input is currently hidden behind something, giving it focus is going to be real confusing. 
Would need some testing and reasoning with more complex situations to assess side effects. 

In the meantime, give the auto focus extension a go. Let's you just forget about chat input focus entirely. 
https://github.com/LenAnderson/SillyTavern-AutoFocus

**Comment by deciare** [comment]
> In the meantime, give the auto focus extension a go. Let's you just forget about chat input focus entirely. https://github.com/LenAnderson/SillyTavern-AutoFocus

That's fantastic, I wasn't aware this existed. That pretty much obviates the need for this pull request.

**Comment by Cohee1207** [comment]
TL;DR for future seekers: Use an extension 
https://github.com/LenAnderson/SillyTavern-AutoFocus
